shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.osv import expression
+
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+from odoo.addons.shopfloor.utils import to_float
+
+
+class Reception(Component):
+    _inherit = "shopfloor.reception"
+
+    def __init__(self, work_context):
+        super().__init__(work_context)
+        self.packaging_update_done = False
+
+    def _before_state__set_quantity(self, picking, line, message=None):
+        """Show the packaging dimension screen before the set quantity screen."""
+        if self.work.menu.set_packaging_dimension and not self.packaging_update_done:
+            packaging = self._get_next_packaging_to_set_dimension(line.product_id)
+            if packaging:
+                return self._response_for_set_packaging_dimension(
+                    picking, line, packaging, message=message
+                )
+        return super()._before_state__set_quantity(picking, line, message=message)
+
+    def _get_domain_packaging_needs_dimension(self):
+        return expression.OR(
+            [
+                [("packaging_length", "=", 0)],
+                [("packaging_length", "=", False)],
+                [("width", "=", 0)],
+                [("width", "=", False)],
+                [("height", "=", 0)],
+                [("height", "=", False)],
+                [("weight", "=", 0)],
+                [("weight", "=", False)],
+                [("qty", "=", 0)],
+                [("qty", "=", False)],
```

Comment:
Discussing with @jbaudoux , we can remove these conditions as qty is now required